controllers/platformnetwork_controller.go

```diff
@@ -420,6 +450,13 @@ func (r *PlatformNetworkReconciler) ReconcileAddressPool(client *gophercloud.Ser
 	} else {
 		if pool == nil {
 			pool, err = r.ReconcileNewAddressPool(client, instance)
+		} else if instance.Spec.Type == cloudManager.MgmtNetworkType || instance.Spec.Type == cloudManager.AdminNetworkType {
+			if _, ok := poolUpdateRequired(instance, pool, r); ok {
+				err = r.ReconciledDeletedAddressPool(client, instance, pool)
```

Comment:
This method attempts to delete an already deleted address pool, and for this, we want to the deletion. We either need to change the name of the reconcile or create a new one for this case (also, update the existing method comment, please).

Reply:
> This method attempts to delete an already deleted address pool, and for this, we want to the deletion. We either need to change the name of the reconcile or create a new one for this case (also, update the existing method comment, please).

Done.